cardinal/events/events.go

```diff
@@ -7,160 +7,83 @@ import (
 	"time"
 
 	"github.com/gofiber/contrib/websocket"
-	"github.com/gofiber/fiber/v2"
 	"github.com/rotisserie/eris"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 )
 
 const shutdownPollInterval = 200
 
-type EventHub interface {
-	EmitEvent(event *Event)
-	FlushEvents()
-	ShutdownEventHub()
-	Run()
-	UnregisterConnection(ws *websocket.Conn)
-	RegisterConnection(ws *websocket.Conn)
-}
-
 const (
 	writeDeadline = 5 * time.Second
-	bufferSize    = 1024
 )
 
-type loggingEventHub struct {
-	logger     *zerolog.Logger
-	eventQueue []*Event
-	running    atomic.Bool
-	broadcast  chan *Event
-	shutdown   chan bool
-	flush      chan bool
-}
-
-func (eh *loggingEventHub) EmitEvent(event *Event) {
-	eh.broadcast <- event
-}
-
-func (eh *loggingEventHub) FlushEvents() {
-	eh.flush <- true
-}
-
-func (eh *loggingEventHub) UnregisterConnection(_ *websocket.Conn) {}
-
-func (eh *loggingEventHub) RegisterConnection(_ *websocket.Conn) {}
-
-func (eh *loggingEventHub) Run() {
-	if eh.running.Load() {
-		return
-	}
-	eh.running.Store(true)
-	for eh.running.Load() {
-		select {
-		case event := <-eh.broadcast:
-			eh.eventQueue = append(eh.eventQueue, event)
-		case <-eh.flush:
-			wg := sync.WaitGroup{}
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-				for _, event := range eh.eventQueue {
-					eh.logger.Info().Msg("EVENT: " + event.Message)
-				}
-			}() // a goroutine is not technically necessary here but this imitates the websocket eventhub as much as possible.
-			wg.Wait()
-			eh.eventQueue = eh.eventQueue[:0]
-		case <-eh.shutdown:
-			eh.running.Store(false)
-		}
-	}
-}
-
-func (eh *loggingEventHub) ShutdownEventHub() {
-	eh.shutdown <- true
+type Event struct {
+	Message string
 }
 
-func NewLoggingEventHub(logger *zerolog.Logger) EventHub {
-	res := loggingEventHub{
-		eventQueue: make([]*Event, 0),
-		running:    atomic.Bool{},
-		broadcast:  make(chan *Event),
-		shutdown:   make(chan bool),
-		flush:      make(chan bool),
-		logger:     logger,
-	}
-	res.running.Store(false)
-	go func() {
-		res.Run()
-	}()
-	return &res
+type EventHub struct {
+	websocketConnections map[*websocket.Conn]bool
+	broadcast            chan *Event
+	flush                chan bool
+	register             chan *websocket.Conn
+	unregister           chan *websocket.Conn
+	shutdown             chan bool
```

Comment:
i actually thing this can be significantly simplified by removing register/unregister entirely given that we realistically will only create a websocket handler once anyway, but i didn't want to it in this PR to avoid blocking Brian

Reply:
do whatever. Looking at the swagger stuff it looks largely unrelated to this.